### PR TITLE
Implement RTD Support

### PR DESCRIPTION
**Pull request by Daltz333**
This PR implements retrieving site URL from ReadTheDocs.

This is enabled by default on RTD platforms and when site URL is not set. Users who set site URL manually will not be overridden, so this is backwards compatible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ extensions = [
 ## Options
 These values are placed in the conf.py of your sphinx project.
 
+Users hosting documentation *do not* need to set any of the following unless custom configuration is wanted. The extension will automatically retrieve your site url.
```
Comment (TheTripleV) [inline README.md:22]: Mention that autopopulation only occurs on Readthedocs.

**Review by TheTripleV — CHANGES_REQUESTED** [review]

**Comment by TheTripleV** [comment]
Ooh thanks for reminding me of test cases.

For test cases:
You can set environment variables with https://docs.pytest.org/en/stable/monkeypatch.html#monkeypatching-environment-variables:~:text=behavior.%22%22%22-,monkeypatch.setenv(%22USER%22%2C%20%22TestingUser%22)
You can probably set html_baseurl and needed rtd vars in conf.py of each test root.

It would be nice if you could include test cases for:
- [ ] RTD ENV is set & html_baseurl is set
- [ ] RTD ENV is set & html_baseurl is NOT set
- [ ] RTD ENV is set & ogp_site_url is set

File: tests/conftest.py [hunk 2]
```diff
@@ -2,9 +2,17 @@
 from bs4 import BeautifulSoup
 from sphinx.testing.path import path
 
+from sphinx.application import Sphinx
+
+
 pytest_plugins = "sphinx.testing.fixtures"
 
 
+def set_config_options(app: Sphinx, config: dict) -> None:
```
Comment (Modelmat) [inline tests/conftest.py:11]: You don't actually end up using this.

**Comment by TheTripleV** [comment]
Fixed the tests (kinda).
```py
urlparse(config["html_baseurl"])
```
doesn't return None; It returns a parse object.

All 3 of the tests pass when run individually but don't work when run together.

**Review by TheTripleV — APPROVED** [review]

**Pull request merged**